src/service/service_layer.cc

```diff
@@ -125,6 +159,50 @@ std::vector<ChirpObj> ServiceLayer::Monitor(const std::string& uname) {
   return chirps;
 }
 
+std::vector<ChirpObj> ServiceLayer::CheckTag(std::vector<ChirpObj> &chirps,const std::string& hash_tag){
+  for(std::vector<ChirpObj>::iterator it = chirps.begin(); it != chirps.end();) {
+      //bool to check if contain hash rag
+      bool contain = false;
+      //text of chirp
+      std::string text = it->text();
+      // holds all the positions that sub occurs within #
+      std::vector<size_t> positions; 
+      size_t pos = text.find("#", 0);
+      while(pos != std::string::npos)
+      {
+          positions.push_back(pos);
+          pos = text.find("#",pos+1);
+      }
+      if(positions.size() == 0){
+        it = chirps.erase(it);
+        continue;
+      }
+
+      //check all positions with #
+      for(auto p:positions){
+        std::string sub = text.substr(p);
+        std::size_t pos1 = sub.find(" ");
+        std::string tag = sub.substr (1,pos1);
+        //trim the tags with white space
+        // tag.erase( std::remove_if( tag.begin(), tag.end(), ::isspace ), tag.end() );
+        if(tag == hash_tag){
+          // std::cout<<tag<<std::endl;
+          // std::cout<<hash_tag<<std::endl;
+          contain = true;
+        }
+      }
+      //if doesnt contain 
+      if(contain == false){
+        it = chirps.erase(it);
+        continue;
```

Comment:
is this necessary?

Reply:
Thanks, I will fix this, I will write a function to check if a chirp contain hashtag